ext/search_kit/Civi/Api4/Action/SearchDisplay/Download.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Civi\Api4\Action\SearchDisplay;
+
+/**
+ * Download the results of a SearchDisplay as a spreadsheet.
+ *
+ * Note: unlike other APIs this action directly outputs a file.
+ *
+ * @package Civi\Api4\Action\SearchDisplay
+ */
+class Download extends AbstractRunAction {
+
+  /**
+   * Requested file format
+   * @var string
+   * @required
+   * @options csv
+   */
+  protected $format;
+
+  /**
+   * @param \Civi\Api4\Generic\Result $result
+   * @throws \API_Exception
+   */
+  protected function processResult(\Civi\Api4\Generic\Result $result) {
+    $entityName = $this->savedSearch['api_entity'];
+    $apiParams =& $this->savedSearch['api_params'];
+    $settings = $this->display['settings'];
+
+    // Displays are only exportable if they have actions enabled
+    if (empty($settings['actions'])) {
+      \CRM_Utils_System::permissionDenied();
+    }
+
+    // Force limit if the display has no pager
+    if (empty($settings['pager']) && !empty($settings['limit'])) {
+      $apiParams['limit'] = $settings['limit'];
+    }
+    $apiParams['orderBy'] = $this->getOrderByFromSort();
+    $this->augmentSelectClause($apiParams);
+
+    $this->applyFilters();
+
+    $apiResult = civicrm_api4($entityName, 'get', $apiParams);
+
+    $rows = $this->formatResult($apiResult);
+
+    $columns = [];
+    foreach ($this->display['settings']['columns'] as $col) {
+      $col += ['type' => NULL, 'label' => '', 'rewrite' => FALSE];
+      if ($col['type'] === 'field' && !empty($col['key'])) {
+        $columns[] = $col;
+      }
+    }
+
+    // This weird little API spits out a file and exits instead of returning a result
+    \CRM_Utils_System::setHttpHeader('Content-Disposition', 'attachment; filename=' . \CRM_Utils_String::munge($this->display['label']) . '.' . $this->format);
```

Comment:
Thanks @colemanw. Noticing this has the same filename problem the pdf ones did that are based on subject. If the user-entered text is in e.g. russian the filename looks like "or5jXD4tUWlbGTHxM9MdStHqkrthNJ123JQXzYOz7b0rTKwJweYtXijJm90aAM7.csv". There's the recently added [CRM_Utils_File::makeFilenameWithUnicode](https://github.com/civicrm/civicrm-core/blob/7420e2ea6c611e0c8a7519d1acaff6f85719832c/CRM/Utils/File.php#L475) which has the same signature as munge().

Reply:
Another good catch @demeritcowboy 
updated to use your suggested function.

Reply:
Sorry this is yet another thing - on the filename, it seems there's more involved. It works with pdf files because it does this extra stuff: https://github.com/dompdf/dompdf/blob/4c65810c797674d0ceb8918859c645f298b5e358/src/Helpers.php#L144. Without that the filename ends up with goofy chars.

Reply:
Ok I updated this PR to use that function.

Reply:
This line ends up ignoring the utf8 part, and then it seems to get further rewritten somewhere after?  `fileName = disp.split(';')[1].split('=')[1].replace(/"/g, '');`
Tricky.